app/controllers/admin/cycles/plugin_relations/petitions_controller.rb

```diff
@@ -1,20 +1,25 @@
 class Admin::Cycles::PluginRelations::PetitionsController < Admin::ApplicationController
   def index
-    @petition = @plugin_relation.petition_information
+    @petition = @plugin_relation.petition_detail
   end
 
   def new
-    @petition = PetitionPlugin::Information.new
+    @petition = PetitionPlugin::Detail.new
   end
 
   def edit
-    @petition = @plugin_relation.petition_information
+    @petition = @plugin_relation.petition_detail
   end
 
   def update
-    @petition = @plugin_relation.petition_information
+    @petition = @plugin_relation.petition_detail
     @petition.update_attributes petition_params
 
+    current_version = @petition.current_version
+    if current_version.nil? || current_version.body != petition_versionable_params[:body]
```

Comment:
IMO this a business logic and shouldn't be here. In cases like these, I prefer using a `use_case` and put this logic there. The controller should only call your `use_case` and get the output.